File: tools/004-clients/Retry.java
// java.net.http.HttpClient に自動再試行の設定は無い。429 をそのまま返すかを測る。
import java.net.URI;
import java.net.http.*;

public class Retry {
    public static void main(String[] args) throws Exception {
        HttpClient c = HttpClient.newHttpClient();
        HttpResponse<String> r = c.send(
            HttpRequest.newBuilder(URI.create(args[0])).build(),
            HttpResponse.BodyHandlers.ofString());
        System.out.println("status=" + r.statusCode()
            + " retry-after=" + r.headers().firstValue("retry-after").orElse("none"));
        System.out.println("java=" + Runtime.version());
    }
}

File: tools/004-clients/retry.go
// net/http の Client に自動再試行は無い。429 をそのまま返すかを測る。
package main

import (
	"fmt"
	"net/http"
	"os"
	"runtime"
)

func main() {
	resp, err := http.Get(os.Args[1])
	if err != nil {
		fmt.Println("error:", err)
		return
	}
	defer resp.Body.Close()
	fmt.Printf("status=%d retry-after=%s\n", resp.StatusCode, resp.Header.Get("Retry-After"))
	fmt.Println("go=" + runtime.Version())
}
